hypervisor/src/arch/x86/mod.rs

```diff
@@ -306,6 +309,24 @@ pub struct MsrEntry {
     pub data: u64,
 }
 
+/// Error that may be returned when attempting to enable AMX state components for guests
+#[derive(Debug, Error)]
+pub enum AmxGuestSupportError {
+    /// Attempted to enable AMX on a CPU vendor that does not support this feature.
+    #[error("The host CPU's vendor does not support AMX features. Only Intel provides such CPUs.")]
+    VendorDoesNotSupportAmx,
```

Comment:
nit: this could hold the current vendor string if it is not too much work to enable it

Reply:
I think I would prefer to just keep it the way it is. This is not something you can solve at runtime and it should be easy enough to figure what vendor the host has outside of CHV.